std/codec.go

```diff
@@ -5,25 +5,9 @@ import (
 	"github.com/cosmos/cosmos-sdk/codec/types"
 	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/types/module"
 	vesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
 )
 
-// ----------------------------------------------------------------------------
-// necessary types and interfaces registered. This codec is provided to all the
-// modules the application depends on.
-//
-// NOTE: This codec will be deprecated in favor of AppCodec once all modules are
-// migrated.
-func MakeCodec(bm module.BasicManager) *codec.LegacyAmino {
```

Comment:
I ran into some nasty import cycles, in tests only:

```
import cycle not allowed in test
package github.com/cosmos/cosmos-sdk/x/auth/tx (test)
    imports github.com/cosmos/cosmos-sdk/std
    imports github.com/cosmos/cosmos-sdk/baseapp
    imports github.com/cosmos/cosmos-sdk/client/grpc/simulate
    imports github.com/cosmos/cosmos-sdk/x/auth/tx
FAIL    github.com/cosmos/cosmos-sdk/x/auth/tx [setup failed]
```

All imports made sense to me, except maybe `std imports baseapp. So I refactored:
- remove mention of `baseapp` from std
- put `consensus_params.go` into params keeper
- remove mention of `types/module` from std (another import cycle)
- this function `MakeCodec` was only used in tests in simapp, so that's where I put it.